src/mprismetadata.cpp

```diff
@@ -390,7 +390,7 @@ QVariantMap MprisMetaDataPrivate::typedMetaData() const
 
 void MprisMetaDataPrivate::setMetaData(const QString &key, const QVariant &value)
 {
-    if (m_metaData.value(key) != value) {
+    if (m_metaData.value(key) != value || m_metaData.contains(key) && (!value.isValid() || value.isNull())) {
```

Comment:
I would write this as:
```
        if (!value.isValid() || value.isNull()) {
                if (!m_metaData.remove(key))
                        return;
        } else if (m_metaData.value(key) != value) {
                m_metaData[key] = value;
        } else {
                return;
        }

        m_changedDelay.start();
```

To avoid duplicated conditions.